src/v/redpanda/admin/debug.cc

```diff
@@ -229,29 +232,29 @@ void admin_server::register_debug_routes() {
       [this](std::unique_ptr<ss::http::request>) {
           vlog(adminlog.info, "Request to get leaders info");
           using result_t = ss::httpd::debug_json::leader_info;
-          std::vector<result_t> ans;
-
+          // TODO(rockwood): get_leaders can lead to a reactor stall, fix to use
+          // an async version
```

Comment:
question: Did you file a core-internal ticket for this?  I can if you like

Reply:
Michal already has a PR that I can use to fix this, just need to wait for https://github.com/redpanda-data/redpanda/pull/16512 (and backport it)